src/coreclr/jit/codegenxarch.cpp

```diff
@@ -491,12 +491,22 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
             if (vecCon->IsAllBitsSet())
             {
-                if ((attr != EA_32BYTE) || compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+                if ((attr == EA_16BYTE) || ((attr == EA_32BYTE) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2)))
```

Comment:
@tannergooding Let me know if this is the right check here

Reply:
Looks correct, but its also going to force an additional check for the common path.

Now that we have 3 types, it might be worth a `switch (attr)` instead.

Reply:
Actually on second thought it might actually miss an EA_8BYTE case for `Vector2`...

I don't recall if we're handling that or not...

If we changed to a `switch` then we can cover an `unreached` path and can also simplify `EA_32BYTE` to just have the `FEATURE_SIMD` one since it should likewise not be encountered otherwise.

Reply:
```
if (vecCon->IsAllBitsSet())
            {
                switch (attr)
                {
                    case EA_8BYTE:
                    case EA_16BYTE:
                    {
                        emit->emitIns_R_R(INS_pcmpeqd, attr, targetReg, targetReg);
                        break;
                    }
#if defined(FEATURE_SIMD)
                    case EA_32BYTE:
                    {
                        if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2))
                            emit->emitIns_SIMD_R_R_R(INS_pcmpeqd, attr, targetReg, targetReg, targetReg);
                        break;
                    }

                    case EA_64BYTE:
                    {
                        assert(compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F));
                        emit->emitIns_SIMD_R_R_R_I(INS_vpternlogd, attr, targetReg, targetReg, targetReg, (int8_t)0xFF);
                        break;
                    }
#endif // FEATURE_SIMD

                    default:
                    {
                        unreached();
                    }
                }
            }
```

Reply:
Yeah...this doesn't seem right. We need to 'break' from the original switch too


```
            if (vecCon->IsAllBitsSet())
            {
                switch (attr)
                {
                    case EA_8BYTE:
                    case EA_16BYTE:
                    {
                        emit->emitIns_R_R(INS_pcmpeqd, attr, targetReg, targetReg);
                        return;
                    }
#if defined(FEATURE_SIMD)
                    case EA_32BYTE:
                    {
                        if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2))
                        {
                            emit->emitIns_SIMD_R_R_R(INS_pcmpeqd, attr, targetReg, targetReg, targetReg);
                            return;
                        }
                        break;
                    }

                    case EA_64BYTE:
                    {
                        assert(compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F));
                        emit->emitIns_SIMD_R_R_R_I(INS_vpternlogd, attr, targetReg, targetReg, targetReg,
                                                   static_cast<int8_t>(0xFF));
                        return;
                    }
#endif // FEATURE_SIMD

                    default:
                    {
                        unreached();
                    }
                }
            }
```